jax/api.py

```diff
@@ -878,20 +865,23 @@ def pmap(fun, axis_name=None, devices=None, backend=None, axis_size=None):
   @wraps(fun)
   def f_pmapped(*args, **kwargs):
     f = lu.wrap_init(fun)
-    args, in_tree = tree_flatten((args, kwargs))
-    local_axis_size = _pmap_axis_size(args)
-    _check_args(args)
+    args_flat, in_tree = tree_flatten((args, kwargs))
+    in_axes_flat = flatten_axes(in_tree, in_axes)
+    local_axis_size = _check_axis_sizes(in_tree, args_flat, in_axes_flat)
+    _check_args(args_flat)
     flat_fun, out_tree = flatten_fun(f, in_tree)
-    out = pxla.xla_pmap(
+    out_flat = pxla.xla_pmap(
         flat_fun,
-        *args,
+        *args_flat,
         backend=backend,
         axis_name=axis_name,
         axis_size=local_axis_size,
         global_axis_size=axis_size,
+        in_dims=in_axes_flat,
```

Comment:
I think that is makes sense to merge this "in_dims" with the newly added "mapped_invars" param. That one is at the moment only added deeper in the code for staged jaxprs, but has overlapping semantics.